src/editing_tools/CropTool.vala

```diff
@@ -41,23 +41,30 @@ public class EditingTools.CropTool : EditingTool {
     private const float MIN_ASPECT_RATIO = 1.0f / 64.0f;
     private const float MAX_ASPECT_RATIO = 64.0f;
 
-    private class ConstraintDescription {
-        public string name;
-        public int basis_width;
-        public int basis_height;
-        public bool is_pivotable;
-        public float aspect_ratio;
+    private class ConstraintDescription : Object {
+        public string name { get; construct; }
+        public int basis_width { get; set construct; }
+        public int basis_height { get; set construct; }
+        public bool is_pivotable { get; construct; }
+        public float new_aspect_ratio { get; construct; }
+        public float aspect_ratio { get; set; }
 
         public ConstraintDescription (string new_name, int new_basis_width, int new_basis_height,
                                       bool new_pivotable, float new_aspect_ratio = COMPUTE_FROM_BASIS) {
-            name = new_name;
-            basis_width = new_basis_width;
-            basis_height = new_basis_height;
-            if (new_aspect_ratio == COMPUTE_FROM_BASIS)
+            Object (
+                name: new_name,
+                basis_width: new_basis_width,
+                basis_height: new_basis_height,
+                is_pivotable: new_pivotable
+            );
+        }
+
+        construct {
+            if (new_aspect_ratio == COMPUTE_FROM_BASIS) {
                 aspect_ratio = ((float) basis_width) / ((float) basis_height);
-            else
+            } else {
                 aspect_ratio = new_aspect_ratio;
-            is_pivotable = new_pivotable;
+            }
```

Comment:
Put it just after `Object` and store the temporary `aspect_ratio` into `new_aspect_ratio` so that the object creation can have something like: `aspect_ratio: new_aspect_ratio`

Reply:
Thank You! I fixed it just now.